libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/barrier.h

```diff
@@ -180,6 +180,40 @@ friend class _CUDA_VSTD::__barrier_poll_tester_parity;
         )
     }
 
+    _LIBCUDACXX_NODISCARD_ATTRIBUTE _LIBCUDACXX_DEVICE
```

Comment:
Is there a practical reason for this to be device only? Since we are already doing a fallback for pre-SM90, why not just use the exact same fallback for the host?

Reply:
Ah, I see; the function is device only because we're still assuming we're in shared memory. I'm still uneasy about that.

Reply:
It's not device only anymore. Host path has been added.